data_analysis/near-deduplication/README.md

```diff
@@ -2,29 +2,211 @@
 
 Code for data near deduplication.
 
-### Setup 
+### Setup
+
 ````
 pip install -r requirements.txt
 ````
 
 Login to be able to be able to push the dataset to the hub after deduplication and clone your huggingface-hub repositories:
+
 ````
 huggingface-cli login
 ````
+
 And make sure you have git-lfs installed.
 
-If you use datasets with different column names from the BigCode ones, you might need to change `PATH_COLUMN` and `CONTENT` variables in `minhash_deduplication.py`. 
+If you use datasets with different column names from the BigCode ones, you might need to change `PATH_COLUMN` and `CONTENT` variables in `minhash_deduplication.py`.
+
+### Usage
 
-### Usage 
 To run near deduplication use the following command and adapt the arguments for your case:
+
 ````
 python near_deduplicate.py \
     --dataset_name bigcode-data/python_any_license_v2 \
     --org bigcode-data \
     --repo_name python_any_license_v2_near_dedup \
     --out_path ./data/any_license-near-dedup \
     --text_column content 
-```` 
-To make just a test run with a subset of the data set `test_run` argument to True. 
+````
+
+To make just a test run with a subset of the data set `test_run` argument to True.
 
 The first time you load the dataset might be slow if it is large, but the data is saved in the cache thanks to `datasets`, and the subsequent calls will be fast.
+
+### Alternative Deduplication Script
+
+`minhash_deduplication_alt.py` is an alternative you might find useful to use as well. It uses `dataset.map` for both hashing and querying as well as `networkx` for duplicate community detection.
+
+I will add an arg parser later, but for now you can update the config in `minhash_deduplication_alt.py` to use the script on custom dataset.
+
+```bash
+pip install -r requirements_alt.txt
+python minhash_deduplication_alt.py
+```
+
+#### Benchmark
+
+Running this script for `codeparrot/codeparrot-clean` on a 10-core machine:
```

Comment:
It would be useful to mention the size of this dataset (50GB) and how this scales for larger datasets (like your comments in the PR)

We could also add an example of execution commands when the argparser is added to the script

Reply:
The arg parser is already added via typer, so it is easy to check the usage. 
```
 Usage: minhash_deduplication_alt.py [OPTIONS]                                                                                                                          
                                                                                                                                                                        
╭─ Options ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --dataset                                     TEXT                             The dataset to use [default: codeparrot/codeparrot-clean-valid]                       │
│ --config                                      TEXT                             Dataset config [default: default]                                                     │
│ --split                                       TEXT                             Dataset split [default: train]                                                        │
│ --data-dir                                    TEXT                             Dataset data directory [default: None]                                                │
│ --column                                      TEXT                             Dataset column [default: content]                                                     │
│ --cache-dir                                   TEXT                             Cache directory [default: .cache]                                                     │
│ --num-perm                                    INTEGER                          Number of permutations [default: 256]                                                 │
│ --seed                                        INTEGER                          Random seed [default: 42]                                                             │
│ --threshold                                   FLOAT                            Minhash threshold [default: 0.85]                                                     │
│ --input-neighbor-dataset                      TEXT                             Resume from a queried dataset [default: None]                                         │
│ --output-neighbor-dataset                     TEXT                             Store a queried dataset [default: None]                                               │
│ --input-graph                                 TEXT                             Resume from a graph [default: None]                                                   │
│ --output-graph                                TEXT                             Store a graph [default: None]                                                         │
│ --input-duplicate-ids                         TEXT                             Resume from computed duplicate ids [default: None]                                    │
│ --output-duplicate-ids                        TEXT                             Store computed duplicate ids [default: None]                                          │
│ --output                                      TEXT                             Store the deduplicated dataset [default: None]                                        │
│ --lfs                             --no-lfs                                     Use LFS files [default: no-lfs]                                                       │
│ --min-token-length                            INTEGER                          Minimum token length [default: 10]                                                    │
│ --community-detection                                                          Use community detection                                                               │
│ --report-false-positive-rate                                                   Report false positive rate based on random samples                                    │
│ --verbose                     -v                                               Verbose logging                                                                       │
│ --install-completion                          [bash|zsh|fish|powershell|pwsh]  Install completion for the specified shell. [default: None]                           │
│ --show-completion                             [bash|zsh|fish|powershell|pwsh]  Show completion for the specified shell, to copy it or customize the installation.    │
│                                                                                [default: None]                                                                       │
│ --help                                                                         Show this message and exit.                                                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```